### ISSUE TITLE
Status of the member based on the payment of invoice

### ISSUE THREAD

**Issue by adriennsm** [Dolibarr/dolibarr#22706]
### Feature Request

To have a new status of the member according to the payment of his subscription if invoiced.

### Use case

I am using the "Members" module and when I register a member and their subscription, I use the functionality to create an invoice without payment (the payment will come later).

However, the status of the member is "Up to date" even though the member has not paid - the invoice is "Unpaid".

### Suggested implementation

In summary:
- creation of the member → Draft
- validation of the member → Validated
- if creating of a subscription fee with invoice creation without payment → Unpaid
- payment of the invoice linked to the subscription fee → Up to date

**Comment by daraelmin** [comment]
Hi @ksar-ksar,
Is this a bug on older version ? It works for me on v17 as described.
Cheers

**Comment by JonBendtsen** [comment]
I wish this issue could be re-opened, because even in Dolibarr v23.0.0 (official docker image) a member is listed with uptodate status even though the invoice is marked abandoned.

**Member list - uptodate**
![Image](https://github.com/user-attachments/assets/09199b08-a03a-4ed9-8077-bbb9829f8a91)

**Member contribution**
![Image](https://github.com/user-attachments/assets/cf2662d4-d64d-4754-9816-38debb71113e)

**Member contribution details**
![Image](https://github.com/user-attachments/assets/7475675d-3578-4c4b-8a57-4bba909666fc)

**Member contribution mouse hover**
![Image](https://github.com/user-attachments/assets/b1e9d309-fdbb-41be-a1e8-2640b17c0b52)

**Invoice with member contribution is abandoned**
![Image](https://github.com/user-attachments/assets/4c30146b-e05c-46b5-ae1b-1c4f464f1265)

**Member contribution details after deleting invoice**
<img width="605" height="698" alt="Image" src="https://github.com/user-attachments/assets/04161fb9-8fd9-461e-a044-f253cbd2659b" />

**Member  status is still uptodate even though invoice has been deleted, the contribution is still recorded as contributed**
![Image](https://github.com/user-attachments/assets/410871c3-57d6-4bb2-974d-9b752c4ca013)
